services/vision/mlvision/detector.go

```diff
@@ -104,11 +104,26 @@ func attemptToBuildDetector(mlm mlmodel.Service, nameMap *sync.Map) (objectdetec
 			return nil, errors.New("output tensor sizes did not match each other as expected")
 		}
 		detections := make([]objectdetection.Detection, 0, len(scores))
+		detectionBoxesAreProportional := false
 		for i := 0; i < len(scores); i++ {
-			xmin, ymin, xmax, ymax := utils.Clamp(locations[4*i+getIndex(boxOrder, 0)], 0, 1)*float64(origW),
-				utils.Clamp(locations[4*i+getIndex(boxOrder, 1)], 0, 1)*float64(origH),
-				utils.Clamp(locations[4*i+getIndex(boxOrder, 2)], 0, 1)*float64(origW),
-				utils.Clamp(locations[4*i+getIndex(boxOrder, 3)], 0, 1)*float64(origH)
+			// heuristic for knowing if bounding box coodinates are abolute pixel locations, or
+			// proportional pixel locations. Absolute bounding boxes will not usually be less than a pixel
+			// and purely located in the upper left corner.
+			if i == 0 && (locations[0]+locations[1]+locations[2]+locations[3] < 4.) {
+				detectionBoxesAreProportional = true
+			}
+			var xmin, ymin, xmax, ymax float64
+			if detectionBoxesAreProportional {
+				xmin = utils.Clamp(locations[4*i+getIndex(boxOrder, 0)], 0, 1) * float64(origW)
```

Comment:
I think these should also be `origW-1` since clamp could return `1` if the model returned an `xmax` / `ymax` that was `1 + ε`, which I don't think is safe to assume is impossible even in proportional mode.